src/Microsoft.AspNetCore.Routing/InvalidRouteException.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public class InvalidRouteException : Exception
```

Comment:
This doesn't seem to properly represent the exception case it's being used for. There's nothing wrong with the *route* as far as I can tell. The route is great. But something else in the system is busted: a particular constraint cannot be created for some unknown reason.